tests/test_cases/flock.c

```diff
@@ -0,0 +1,77 @@
+#undef _GNU_SOURCE
+#define _GNU_SOURCE
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/fcntl.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+// Helper function to remove a file
+int remove_file(char file_name[]) {
+  int rm = remove(file_name);
+  if (rm == -1) {
+    perror("FAILED TO REMOVE FILE\n");
+    exit(EXIT_FAILURE);
+  }
+  return 0;
+}
+
+int main(int argc, char **argv) {
+  // Create a temporary file to use
+  char file_name[100] = "testfiles/flock.txt";
```

Comment:
Using `#define` to store a path is a recommended practice as it enhances code readability.